tests/test_upload.py

```diff
@@ -18,6 +18,11 @@
     "local_path2.fastq.gz": "gcs/path/4321/fastq/2019-09-04T18:59:45.224099",
 }
 EXTRA_METADATA = {"lp1": "uuid1"}
+GCS_FILE_MAP = {
+    "gcs/path/1234/fastq/2019-09-04T18:59:45.224099": "uuid2",
+    "gcs/path/4321/fastq/2019-09-04T18:59:45.224099": "uuid3",
+}
+OPTIONAL_FILES = []
```

Comment:
I think it'd be good to add a test where there are missing optional files, too, to illustrate the functionality you're adding in this PR.

Reply:
in next commit